### DISCUSSION TITLE
POST should always return a response

### DISCUSSION THREAD

**Discussion by johnkmzhou** [directus/directus#20006] [Feature Requests]
### Summary

I know that this has been discussed in the past #18108 but the current situation where a no-read permission collection returns a 204 might be ok in most scenarios but makes working with the `/files` endpoint very challenging.

### Motivation

I have a public registration form for an event. This form requires a file to be uploaded to a folder containing files that should not be viewed publicly. The problem is that if I create a custom filter to prevent these files from being read, it will cause the file upload to respond with a 204, and I won't be able to retrieve the file id that I need to create an item in my registration collection. Here is a sample code snippet:

```
const formData = new FormData();
formData.append("folder", folderId);
formData.append("file", file);
// public has no read permission for this folder so this will return nothing
const fileRes = await restClient.request(uploadFiles(formData));

// fileRes.id is undefined
await restClient.request(
  createItem("registration", { ...formValues, upload: fileRes.id })
);
```

### Detailed Design

For the sake of consistency I personally believe that all posts should return a response no matter what the read permission is. Since for the most part you're returning the same data that the user submitted it shouldn't cause real security concerns. And for fields that you don't want returned, you can always create a flow to control the response.

Another alternative is to carve out an exception for `/files` to always return a response on a POST. This would make the API less consistent than the above solution. You also won't be able to create a flow for this because this feature hasn't been completed: #15762.

### Must Have:
- A POST request to `/files` should always return a response.

### Should Have:
- Documentation updated regarding the new behavior.

### Drawbacks

Documentation would have to be written to inform developers of this new behavior. In some rare cases, a flow would need to be created if they need to filter out response fields.

### Alternatives

There was a recent [suggestion](https://github.com/directus/directus/discussions/8618#discussioncomment-7256282) that a developer could create a custom file upload endpoint. But that seems rather involved rather than just having the POST return a response.

I would argue that if the scenario to provide public permissions to specific files or file folders is common enough that there's a note in the [Directus documentation](https://docs.directus.io/reference/files.html#file-security) for it, then the scenario where a user uploads a file to a private folder is common enough that we shouldn't have to force developers to write a custom endpoint for it.

### Adoption Strategy

For the most part, if a developer wasn't handling a response before due to no-read permission, they can continue doing so after the change unless they feel that some of the fields should not be returned. In that case they can create a flow to filter out the response. But I suspect that is an unlikely scenario.

**Comment by rijkvanzanten** [comment]
My preferred solution here would be to rely on https://github.com/directus/directus/discussions/11464 so you can configure the read permissions for reading a single item by a known ID, which in turn would be used for the read permissions for the single item you just uploaded. That should solve for the list concern and prevents from having to open up all posts to a response with potentially sensitive data

**Comment by rijkvanzanten** [comment]
Heya! Thank you for taking the time to submit this request! 

It has been over **90 days**, and this discussion has not received at least **15 votes** from the community. This means that we don't feel like there's enough community interest to warrant further R&D into this topic at this time. :ice_cube:

This request will now be closed to keep our discussions tidy. Please reach out if you have any questions!

_For more information, see our [Feature Request Process](https://docs.directus.io/contributing/feature-request-process.html)._
